src/BaseHostPlugin.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Allow to modify the the schema and the host of an existing request.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class BaseHostPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $baseUri;
+
+    /**
+     * @var bool
+     */
+    private $replace;
+
+    /**
+     * Available options for $config are:
+     *  - replace: bool True will replace all hosts, false will only add host when none is specified.
+     *
+     * @param UriInterface $baseUri
+     * @param array        $config
```

Comment:
Look at https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#examples-12 on how the intent is for config array parameters in phpdoc, even if the standard is not yet accepted the format for config is pretty stable